README.md

```diff
@@ -1,31 +1,118 @@
+
 ```
-python sample_structure.py --bulk_db bulk_database.pkl --adsorbate_db adsorbate_database.pkl --output_dir outputs/ --seed 0
+python sample_structure.py --bulk_db bulk_db_nelems.pkl \
+  --adsorbate_db adsorbate_db.pkl  \
+  --output_dir outputs/ --seed 0
 ```
 
-2. Generate one specified adsorbate, one more more specified bulks, and all possible surfaces and configs. The following example generates files for all adslab placements of adsorbate 10 on all possible surfaces from bulk 20. Files will be stored in `outputs/10_20_0/surface/`, `outputs/10_20_0/adslab0/`, `outputs/10_20_0/adslab1/`, ..., `outputs/10_20_1/surface/`, `outputs/10_20_1/adslab0/`, and so on for all combinations of surfaces and adslab configs. You may also choose multiple bulks (and all of their surfaces and adslab configs) by giving a comma separated list of bulk indices.
+1. Generate one specified adsorbate, one more more specified bulks, and all possible surfaces and configs. The following example generates files for all adsorbate placements of adsorbate 10 on all possible surfaces from bulk 20. Files will be stored in `outputs/10_20_0/surface/`, `outputs/10_20_0/adslab0/`, `outputs/10_20_0/adslab1/`, ..., `outputs/10_20_1/surface/`, `outputs/10_20_1/adslab0/`, and so on for all combinations of surfaces and adsorbate+surface configs. You may also choose multiple bulks (and all of their surfaces and adsorbate+surface configs) by giving a comma separated list of bulk indices.
+
+```
+python sample_structure.py --bulk_db bulk_db_flat.pkl \
+  --adsorbate_db adsorbate_db.pkl \
+  --output_dir outputs/  \
+  --enumerate_all_structures \
+  --adsorbate_index 10 --bulk_indices 20
 ```
-python sample_structure.py --bulk_db bulk_database.pkl --adsorbate_db adsorbate_database.pkl --output_dir outputs/ --enumerate_all_structures --adsorbate_index 10 --bulk_indices 20
+
+1. Generate one specified adsorbate, one or more specified bulks, one specified surface, and all possible configs. This is the same as #2 except only one surface is selected, which can be used to help parallelize #2. The following example generates files for all adsorbate placements of adsorbate 10 on surface 0 from bulk 20, resulting in files in `outputs/10_20_0/surface/`, `outputs/10_20_0/adslab0/`, `outputs/10_20_0/adslab1/`, and so on for all the adsorbate+surface configs.
+
 ```
+python sample_structure.py --bulk_db bulk_db_flat.pkl \
+  --adsorbate_db adsorbate_db.pkl --output_dir outputs/ 
+  --enumerate_all_structures --adsorbate_index 10 
+  --bulk_indices 20 --surface_index 0
+```
+
+For any of the above, `--precomputed_structures dir/`  can be added to use the precomputed surfaces rather than calculating all possible surfaces of a given bulk from scratch. Adding `--verbose` will print out additional info.
+
+
+## Generating inputs for OC20
```

Comment:
probably should have put the previous comment here. Some sort of instruction on converting what users know (mpid, adsorbate smile string) to get the corresponding index.

Reply:
Good point, uploaded the mapping files and added the links.